commands/search.js

```diff
@@ -14,6 +14,44 @@ class Search extends Command {
     });
   }
 
+  formCatchDescription(pokeBalls, energy, settings) {
+    if (pokeBalls.length === 0) {
+      return { allowCapture: false, description: `You have ${energy-1} energy remaining.\nSadly, you don't have any PokeBalls left, so you have no choice but to let this one run away.\n\n\`${settings.prefix}search\` to continue looking (1 energy)\n\`${settings.prefix}cancel\` to stop searching` };
+    }
+
+    let catchDesc;
+    if (pokeBalls.length > 1) {
+      const otherDesc = pokeBalls.slice(1).map(x => `\`${settings.prefix}catch ${x.name}\` to use your ${x.name},`).join('\n');
+      catchDesc = `${otherDesc}\nor type just \`${settings.prefix}catch\` to use your ${pokeBalls[0].name}.\n`;
+    } else {
+      catchDesc = `Type \`${settings.prefix}catch\` to use your ${pokeBalls[0].name}.`;
+    }
+
+    return { allowCapture: true, description: `You have ${energy-1} energy remaining.\n${catchDesc}\n\`${settings.prefix}search\` to let this blob run away and continue looking (1 energy)\n\`${settings.prefix}cancel\` to let the blob run away and stop searching` };
+  }
+
+  async waitForCatchResponse(message, pokeBalls, escapedPrefix) {
+    const transform = s => s.toLowerCase().replace(/ /g, '');
+    const pokeBallNames = pokeBalls.map(x => transform(x.name)).concat(['']);
+
+    const re = new RegExp(`^${escapedPrefix}(catch|cancel|search)(.*)$`);
+    const filter = m => (m.author.id === message.author.id && re.test(m.content) && pokeBallNames.includes(transform(re.exec(m.content)[2])));
+
+    let response;
+    try {
```

Comment:
I'm pretty sure there is a better way to handle this then a try/catch, but I'm not sure

Reply:
The [discord.js documentation on awaitMessages](https://discord.js.org/#/docs/main/master/class/TextChannel?scrollTo=awaitMessages) shows using ` .catch` on a Promise. In async/await syntax, this [throws an exception so we have to catch it.](https://javascript.info/async-await#error-handling)

Reply:
Alright, fair